jme3-plugins/src/xml/java/com/jme3/export/xml/DOMInputCapsule.java

```diff
@@ -1245,9 +1245,16 @@ public IntMap<? extends Savable> readIntSavableMap(String name, IntMap<? extends
                                 if (n instanceof Element && n.getNodeName().equals("MapEntry")) {
                                         Element elem = (Element) n;
                                         currentElem = elem;
-                                        int key = Integer.parseInt(currentElem.getAttribute("key"));
-                                        Savable val = readSavable("Savable", null);
-                                        ret.put(key, val);
+                                        try {
+	                                        int key = Integer.parseInt(currentElem.getAttribute("key"));
+	                                        Savable val = readSavable("Savable", null);
+	                                        ret.put(key, val);
+                                        }catch (NumberFormatException nfe) {
+                                            IOException io = new IOException(nfe.toString());
+                                            io.initCause(nfe);
```

Comment:
I don't think it's a good idea to throw a checked exception from an unchecked exception, if you want to propagate an unchecked runtime exception to the user application use a `throws NumberFormatException` to give a clue to catch the exception or else use a strategy pattern with a compatible JDK interface to handle this exception from the user application, but this shouldn't throw a business checked exception.